srcmedia/ts/members-search.ts

```diff
@@ -0,0 +1,82 @@
+import { merge, partition } from 'rxjs'
+import { pluck, map, withLatestFrom, startWith } from 'rxjs/operators'
+
+import { RxTextInput, RxCheckboxInput } from './lib/input'
+import { RxOutput } from './lib/output'
+import { RxSearchForm } from './lib/form'
+import { RxSelect } from './lib/select'
+import PageControls from './components/PageControls'
+
+document.addEventListener('DOMContentLoaded', () => {
+
+    /* ELEMENTS */
+    const $membersSearchForm = document.getElementById('members-form') as HTMLFormElement
+    const $keywordInput = document.querySelector('input[name=query]') as HTMLInputElement
+    const $hasCardInput = document.querySelector('input[name=has_card]') as HTMLInputElement
+    const $resultsOutput = document.querySelector('output[form=members-form]') as HTMLOutputElement
+    const $totalResultsOutput = document.querySelector('output.total-results') as HTMLOutputElement
+    const $pageSelect = document.querySelector('select[name=page]') as HTMLSelectElement
+    const $sortSelect = document.querySelector('select[name=sort]') as HTMLSelectElement
+    const $pageControls = document.getElementsByClassName('sort-pages')[0] as HTMLElement
+    const $nextPageLink = document.querySelector('.sort-pages a[rel=next]') as HTMLAnchorElement
+    const $prevPageLink = document.querySelector('.sort-pages a[rel=prev]') as HTMLAnchorElement
+
+    /* COMPONENTS */
+    const membersSearchForm = new RxSearchForm($membersSearchForm)
+    const keywordInput = new RxTextInput($keywordInput)
+    const hasCardInput = new RxCheckboxInput($hasCardInput)
+    const resultsOutput = new RxOutput($resultsOutput)
+    const totalResultsOutput = new RxOutput($totalResultsOutput)
+    const pageSelect = new RxSelect($pageSelect)
+    const sortSelect = new RxSelect($sortSelect)
+    const pageControls = new PageControls($pageControls)
+
+    /* BINDINGS */
+
+    // Sort bindings
+    const [keywordOn, keywordOff] = partition(
```

Comment:
I believe I get what's going on here, but a bit of clarification about the expected behavior would be appreciated.

Reply:
thanks for calling this out - I added some comments to one of these other binding blocks but neglected it here. will add some explanation and a docs link for `partition()` since it's pretty rxjs-specific.

Reply:
@thatbudakguy I think you're probably right, you shouldn't need to document library functionality in the code - let's figure out how to get the rest of the team more up to speed, or have a team/project specific reference somewhere else to help with this.